var/spack/repos/builtin/packages/cudnn/package.py

```diff
@@ -32,9 +32,13 @@ class Cudnn(Package):
 
     homepage = "https://developer.nvidia.com/cudnn"
 
-    version('6.0', 'a08ca487f88774e39eb6b0ef6507451d',
+    version('7.0.cuda9', '6189e05077dbce0bd07059b20306a836',
+            url='http://developer.download.nvidia.com/compute/redist/cudnn/v7.0.3/cudnn-9.0-linux-x64-v7.tgz')
+    version('7.0.cuda8', '8548a35c8d51c3f8d44b46a79026d18b',
+            url='http://developer.download.nvidia.com/compute/redist/cudnn/v7.0.3/cudnn-8.0-linux-x64-v7.tgz')
+    version('6.0.cuda8', 'a08ca487f88774e39eb6b0ef6507451d',
             url='http://developer.download.nvidia.com/compute/redist/cudnn/v6.0/cudnn-8.0-linux-x64-v6.0.tgz')
-    version('5.1', '406f4ac7f7ee8aa9e41304c143461a69',
+    version('5.1.cdua8', '406f4ac7f7ee8aa9e41304c143461a69',
             url='http://developer.download.nvidia.com/compute/redist/cudnn/v5.1/cudnn-8.0-linux-x64-v5.1.tgz')
 
     depends_on('cuda@8:')
```

Comment:
Shouldn't this be made conditional based on which patch version is selected?

What we should really do here is to add a `when` clause to the versions and allow them to be conditional on the dependency version, so you could have:

```python
    version('7.0', 'abc123', url='http://example.com/cudnn/v7.0.3/cudnn-9.0-blah.tar.gz', when='^cuda@9')
    version('7.0', 'xyz123', url='http://example.com/cudnn/v7.0.3/cudnn-8.0-blah.tar.gz', when='^cuda@8')
```

Then the user would just write, e.g., `spack install cudnn@7.0 ^cuda@8`.  Unfortunately, that requires some backward-inference that our current dependency resolution algorithm doesn't have.  I think we'll be able to support this within a year, though.

For now, using @alalazo's trick along with a conditional CUDA dependency should work, I think -- but you should put the conditional dependency so that the user doesn't have to write `spack install cudnn@x.y.cuda8 ^cuda@8` or `spack install cudnn@x.y.cuda9 ^cuda@9`.  The conditional dependency will remove the need for the `^cuda@X`

Reply:
I changed it to 

```
version('7.0', 'abc123', url='http://example.com/cudnn/v7.0.3/cudnn-9.0-blah.tar.gz', when='^cuda@9')
```